examples/light-switch-app/efr32/src/binding-handler.cpp

```diff
@@ -75,96 +83,283 @@ void ProcessOnOffBindingCommand(CommandId commandId, const EmberBindingTableEntr
+    auto onSuccess = [](const ConcreteCommandPath & commandPath, const StatusIB & status, const auto & dataResponse) {
+        ChipLogProgress(NotSpecified, "OnOff command succeeds");
+    };
+
+    auto onFailure = [](CHIP_ERROR error) {
+        ChipLogError(NotSpecified, "OnOff command failed: %" CHIP_ERROR_FORMAT, error.Format());
+    };
+
+    auto onDone = [](CommandSender * commandSender) { ChipLogError(NotSpecified, "OnOff command done"); };
 
-    if (binding.type == EMBER_MULTICAST_BINDING)
+    NodeId sourceNodeId = Server::GetInstance().GetFabricTable().FindFabricWithIndex(binding.fabricIndex)->GetNodeId();
+    Messaging::ExchangeManager & exchangeMgr = Server::GetInstance().GetExchangeManager();
+
+    switch (commandId)
     {
-        ChipLogError(NotSpecified, "Group binding is not supported now");
+    case Clusters::OnOff::Commands::Toggle::Id:
+        Clusters::OnOff::Commands::Toggle::Type toggleCommand;
+        Controller::InvokeGroupCommandRequest(&exchangeMgr, binding.fabricIndex, binding.groupId, sourceNodeId, toggleCommand,
+                                              onSuccess, onFailure, onDone);
+        break;
+
+    case Clusters::OnOff::Commands::On::Id:
+        Clusters::OnOff::Commands::On::Type onCommand;
+        Controller::InvokeGroupCommandRequest(&exchangeMgr, binding.fabricIndex, binding.groupId, sourceNodeId, onCommand,
+                                              onSuccess, onFailure, onDone);
+
+        break;
+
+    case Clusters::OnOff::Commands::Off::Id:
+        Clusters::OnOff::Commands::Off::Type offCommand;
+        Controller::InvokeGroupCommandRequest(&exchangeMgr, binding.fabricIndex, binding.groupId, sourceNodeId, offCommand,
+                                              onSuccess, onFailure, onDone);
+        break;
+
+    default:
+        ChipLogError(NotSpecified, "Invalid binding command data - commandId is not supported");
+        break;
     }
-    else if (binding.type == EMBER_UNICAST_BINDING && binding.local == 1 &&
-             (!binding.clusterId.HasValue() || binding.clusterId.Value() == data->clusterId))
-    {
+}
 
-        switch (data->clusterId)
+void LightSwitchChangedHandler(const EmberBindingTableEntry & binding, DeviceProxy * peer_device, void * context)
+{
+    VerifyOrReturn(context != nullptr, ChipLogError(NotSpecified, "OnDeviceConnectedFn: context is null"));
+    BindingCommandData * data = static_cast<BindingCommandData *>(context);
+
+    if (binding.local == 1 && (!binding.clusterId.HasValue() || binding.clusterId.Value() == data->clusterId))
+    {
+        if (binding.type == EMBER_MULTICAST_BINDING && data->isGroup)
+        {
+            switch (data->clusterId)
+            {
+            case Clusters::OnOff::Id:
+                ProcessOnOffGroupBindingCommand(data->commandId, binding);
+                break;
+            default:
+                ChipLogError(NotSpecified, "Invalid binding command data - clusterId is not supported");
```

Comment:
This is unfortunately too late to discover this, when in fact, we should be rejecting the original write request that added the Binding entry.

This is not something to be fixed in this PR, will have to discuss with @bzbarsky-apple on a potential solution there.

Reply:
done